app/assets/javascripts/models/beta/iteration.js

```diff
@@ -13,11 +13,25 @@ const getIterationForDate = (date, project) => {
     : iterationQuantity + 1;
 };
 
+const getNextIterationNumber = (sprints) => {
+  const lastSprint = sprints[sprints.length - 1];
+
+  return lastSprint ? lastSprint.number + 1 : 1;
+}
+
 export const getDateForIterationNumber = (iterationNumber, project) => {
-  return moment(new Date(project.startDate))
-    .startOf("isoWeek")
-    .add(iterationNumber, "weeks")
-    .format("ddd MMM Do Y");
+  let startDate;
```

Comment:
I think you can break this function in two:
```js
const getStartDate = (iterationNumber, project) => {
  const momentDate = moment(project.startDate);

  if (iterationNumber === 1) {
    return moment(project.startDate);
  } else {
    const missingDaysFromFirstSprint = (momentDate.isoWeekday() - project.iterationStartDay) % 7;
    const iterationDays = (iterationNumber - 1) * (project.iterationLength * 7);

    return (momentDate.subtract(missingDaysFromFirstSprint, 'days')).add(iterationDays, 'days');
  }
}

export const getDateForIterationNumber = (iterationNumber, project) => {
  return getStartDate(iterationNumber, project).format("ddd MMM Do Y");
};
```

Reply:
I agree. Also you dont need `else` when you have a `return` inside the `if` case 😄

Reply:
sure :smile: